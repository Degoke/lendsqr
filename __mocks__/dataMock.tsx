import UsersIcon from '../public/images/users_card_icon.svg'

export const mockCard = {
  image: UsersIcon,
  text: 'Users',
  amount: 100,
}

export const mockImage = UsersIcon

export const MockComponent = () => (
  <div data-testid="mockComponent">Hello World</div>
)

export const mockList = {
  title: 'settings',
  items: [
    {
      text: 'preferences',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'fees and pricing',
      url: '',
      icon: UsersIcon,
    },
    {
      text: 'audit logs',
      url: '',
      icon: UsersIcon,
    },
  ],
}
