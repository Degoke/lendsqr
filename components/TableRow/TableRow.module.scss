.tableRow {
  font-size: 14px;

  &__moreButton {
    position: relative;
  }
}
