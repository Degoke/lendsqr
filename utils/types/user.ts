export interface IUser {
  id: string
  createdAt: string
  orgName: string
  userName: string
  email: string
  phoneNumber: string
  lastActiveDate: string
  profile: UserProfile
  guarantor: UserGuarantor
  accountBalance: string
  accountNumber: string
  socials: UserSocials
  education: UserEducation
  status?: UserStatus
}

export type UserStatus = 'inactive' | 'pending' | 'blacklisted' | 'active'

interface UserProfile {
  firstName: string
  lastName: string
  phoneNumber: string
  avatar: string
  gender: string
  bvn: string
  address: string
  currency: string
}

interface UserGuarantor {
  firstName: string
  lastName: string
  phoneNumber: string
  gender: string
  address: string
}

interface UserSocials {
  facebook: string
  instagram: string
  twitter: string
}

interface UserEducation {
  level: string
  employmentStatus: string
  sector: string
  duration: string
  officeEmail: string
  monthlyIncome: string[]
  loanRepayment: string
}

export type UserFilters =
  | 'orgName'
  | 'email'
  | 'userName'
  | 'createdAt'
  | 'status'
  | 'phoneNumber'
