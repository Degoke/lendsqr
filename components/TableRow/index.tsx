import { IUser } from '../../utils/types/user'
import Pill from '../Pill'
import MoreIcon from '../../public/images/more_icon.svg'
import ViewIcon from '../../public/images/view_icon.svg'
import ActivateIcon from '../../public/images/activate_icon.svg'
import BlacklistIcon from '../../public/images/blacklist_icon.svg'
import Image from 'next/image'
import { useState } from 'react'
import StatusMenu, { StatusMenuItem } from '../StatusMenu'
import IconButton from '../IconButton'
import styles from './TableRow.module.scss'
import { useAppDispatch } from '../../utils/hooks/useRedux'
import { updateCurrentStatusMenu } from '../../features/helpers/helperSlice'
import { activateUser, blacklistUser } from '../../features/Users/userSlice'

type Props = {
  user: IUser
}

export default function TableRow({ user }: Props) {
  const dispatch = useAppDispatch()

  const openMenu = () => {
    dispatch(updateCurrentStatusMenu(user?.id))
  }

  const handleBlacklistUser = () => {
    dispatch(blacklistUser(user?.id))
    dispatch(updateCurrentStatusMenu(''))
  }

  const handleActivateUser = () => {
    dispatch(activateUser(user?.id))
    dispatch(updateCurrentStatusMenu(''))
  }

  const items: StatusMenuItem[] = [
    {
      image: ViewIcon,
      text: 'View Details',
      isLink: true,
    },
    {
      image: BlacklistIcon,
      text: 'Blacklist User',
      isLink: false,
      onClick: handleBlacklistUser,
    },
    {
      image: ActivateIcon,
      text: 'Activate User',
      isLink: false,
      onClick: handleActivateUser,
    },
  ]

  return (
    <>
      <tr>
        <td>{user?.orgName}</td>
        <td>{user?.userName}</td>
        <td>{user?.email}</td>
        <td>{user?.phoneNumber}</td>
        <td>{new Date(user?.createdAt).toDateString()}</td>
        <td>
          <Pill type={user?.status ?? 'pending'} />
        </td>
        <td className={styles.tableRow__moreButton}>
          <IconButton
            image={MoreIcon}
            alt={'more actions'}
            onClick={openMenu}
          />
          <StatusMenu items={items} id={user?.id} />
        </td>
      </tr>
    </>
  )
}
