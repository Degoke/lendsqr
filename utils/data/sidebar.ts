import { urls } from '../constants/urls'
import { IListItem } from '../types/list'
import Users from '../../public/images/users.svg'
import Guarantors from '../../public/images/guarantors.svg'
import Loans from '../../public/images/loans.svg'
import Decision from '../../public/images/decision.svg'
import Savings from '../../public/images/savings.svg'
import Whitelist from '../../public/images/whitelist.svg'
import Karma from '../../public/images/karma.svg'
import Products from '../../public/images/products.svg'
import Fees from '../../public/images/fees.svg'
import Transactions from '../../public/images/transactions.svg'
import Services from '../../public/images/services.svg'
import Service from '../../public/images/service.svg'
import Settlements from '../../public/images/settlements.svg'
import Reports from '../../public/images/reports.svg'
import Pricing from '../../public/images/pricing.svg'
import Audit from '../../public/images/audit.svg'
import Requests from '../../public/images/requests.svg'
import Organization from '../../public/images/organization.svg'
import Preferences from '../../public/images/preferences.svg'

interface ISideBarData {
  title: string
  items: IListItem[]
}

export const sidebarData: ISideBarData[] = [
  {
    title: 'customers',
    items: [
      {
        text: 'users',
        url: urls.USERS,
        icon: Users,
      },
      {
        text: 'guarantors',
        url: '',
        icon: Guarantors,
      },
      {
        text: 'loans',
        url: '',
        icon: Loans,
      },
      {
        text: 'decision models',
        url: '',
        icon: Decision,
      },
      {
        text: 'savings',
        url: '',
        icon: Savings,
      },
      {
        text: 'loan requests',
        url: '',
        icon: Requests,
      },
      {
        text: 'whitelist',
        url: '',
        icon: Whitelist,
      },
      {
        text: 'karma',
        url: '',
        icon: Karma,
      },
    ],
  },
  {
    title: 'businesses',
    items: [
      {
        text: 'organization',
        url: '',
        icon: Organization,
      },
      {
        text: 'loan products',
        url: '',
        icon: Loans,
      },
      {
        text: 'savings products',
        url: '',
        icon: Products,
      },
      {
        text: 'fees and charges',
        url: '',
        icon: Fees,
      },
      {
        text: 'transactions',
        url: '',
        icon: Transactions,
      },
      {
        text: 'services',
        url: '',
        icon: Services,
      },
      {
        text: 'service accounts',
        url: '',
        icon: Service,
      },
      {
        text: 'settlements',
        url: '',
        icon: Settlements,
      },
      {
        text: 'reports',
        url: '',
        icon: Reports,
      },
    ],
  },
  {
    title: 'settings',
    items: [
      {
        text: 'preferences',
        url: '',
        icon: Preferences,
      },
      {
        text: 'fees and pricing',
        url: '',
        icon: Pricing,
      },
      {
        text: 'audit logs',
        url: '',
        icon: Audit,
      },
    ],
  },
]
