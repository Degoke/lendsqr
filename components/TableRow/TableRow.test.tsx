import { render, screen } from '@testing-library/react'
import TableRow from './index'
import { mockCard, mockImage } from '../../__mocks__/dataMock'
import { mockUser } from '../../utils/data/user'
import { mockProvider } from '../../jest.setup'

describe('TableRow tests', () => {
  it('renders TableRow component unchanged', () => {
    const { container } = render(mockProvider(<TableRow user={mockUser[0]} />))
    expect(container).toMatchSnapshot()
  })
})
