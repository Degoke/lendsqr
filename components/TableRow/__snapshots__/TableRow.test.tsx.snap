// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow tests renders TableRow component unchanged 1`] = `
<div>
  <tr>
    <td>
      labore-dolor-et
    </td>
    <td>
      Wilburn.Rice
    </td>
    <td>
      Maverick.Hyatt83@gmail.com
    </td>
    <td>
      (553) 208-0727 x31321
    </td>
    <td>
      Tue Dec 27 2072
    </td>
    <td>
      <div
        class="pill pending"
      >
        <p>
          pending
        </p>
      </div>
    </td>
    <td
      class="tableRow__moreButton"
    >
      <button
        class="iconButton"
      >
        <img
          alt="more actions"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
      </button>
      <div
        class="popover popover"
      />
    </td>
  </tr>
</div>
`;
